crashlog/crashutils.c

```diff
@@ -178,6 +180,33 @@ void do_last_kmsg_copy(char *dir) {
         snprintf(destion, sizeof(destion), "%s/%s", dir, FTRACE_RAMOOPS_FILE);
         do_copy_tail(FTRACE_RAMOOPS, destion, MAXFILESIZE);
     }
+    source = find_system_last_kmsg();
```

Comment:
It's better to do memory allocate and free inside one function. So suggest to not return pointer for find_system_last_kmsg. You can allocate pointer in do_last_kmsg_copy, then pass it to find_system_last_kmsg as parameter. find_system_last_kmsg just needs to return success or fail.